### DISCUSSION TITLE
Welcome to XCharts Discussions!

### DISCUSSION THREAD

**Discussion by monitor1394** [XCharts-Team/XCharts#106] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Hoangndbk** [comment]
Please add the candlestick chart feature, Thanks <3

**Reply by monitor1394** [reply]
Thanks for your suggestion, I will consider supporting it, maybe in version 2.1 or 2.2.

**Reply by Hoangndbk** [reply]
Great, thanks for your support !

**Comment by JimZeeKing** [comment]
Hi it's me again :)

Since i don't want to use issue tracking to ask question, i'll ask them here.

Do you know if it's possible do have this with v2.0? An y axis that can be defined as a visual threshold

![image](https://user-images.githubusercontent.com/19315887/116305139-5e4a4200-a771-11eb-9d20-dccd84c882bf.png)

And also is it possible to create this type of chart with the v2.0 lib ?

![image](https://user-images.githubusercontent.com/19315887/116305316-9c476600-a771-11eb-8b22-f77d9f591e29.png)

Thank you in advance!

**Reply by monitor1394** [reply]
> Do you know if it's possible do have this with v2.0? An y axis that can be defined as a visual threshold

Not support yet.

> And also is it possible to create this type of chart with the v2.0 lib ?

You can use BarChart, but don't support scrollview.

**Reply by JimZeeKing** [reply]
> You can use BarChart, but don't support scrollview.

Is the zoom feature suppose to work with this kind of tilted chart?

**Comment by JimZeeKing** [comment]
Is it possible with a pie chart to have a legend that include percentage with the data label? I think maybe formatter could help but i cannot make it happen. Thanks!

**Reply by monitor1394** [reply]
It is not supported yet.

**Reply by JimZeeKing** [reply]
ok thank you

**Comment by chenkang821711426** [comment]
饼图能否支持3D饼图

**Reply by monitor1394** [reply]
暂无这个计划。

**Reply by chenkang821711426** [reply]
好的，谢谢

**Reply by Bian-Sh** [reply]
已经在 Pro 版本中支持

**Comment by Alinccc** [comment]
能否使用自定义图片来替换掉ItemSytle，
![image](https://user-images.githubusercontent.com/5468346/136930424-5ad32a48-bec3-4ea5-8003-fcb10326862b.png)
每个柱形图都是一张贴图。
![image](https://user-images.githubusercontent.com/5468346/136930720-40c966f1-7cba-4d38-8d45-2dab2c3abb75.png)
这样能做出样式更好看的柱形图。

**Reply by Alinccc** [reply]
![image](https://user-images.githubusercontent.com/5468346/136931006-e09643de-6a5d-4b27-a698-c0a2affca6fa.png)
最好还能设置每个item的背景。

**Reply by monitor1394** [reply]
有付费图表包

PictorialBarChart：
<img width="781" alt="image" src="https://user-images.githubusercontent.com/1450329/137039055-86b15dfd-77d5-44f7-95b6-c893713eafc3.png">

BarChart3D：
<img width="781" alt="image" src="https://user-images.githubusercontent.com/1450329/137039168-2d6e3d1a-73e8-445f-9a28-ee7a28a4ce6d.png">

需要的话email或加Q群私聊

**Comment by Zebra809193074** [comment]
你好，我使用的是最新的2.4.0版本，当我启用TextmeshPro的时候，脚本会提示无法识别TextmeshPro的内容，包括TMPro的命名空间和相关组件，但是工程已经安装好TextmeshPro，在其他地方也能正常使用

**Reply by monitor1394** [reply]
要启动TextmeshPro有两个点：

- 一是在 Project Setting 加上自定义宏 dUI_TextMeshPro
- 二是 XCharts 的相关 asmdef 加上 TextMeshPro 的引用

正常操作的话按[教程：5分钟上手XCharts](https://github.com/monitor1394/unity-ugui-XCharts/blob/master/Doc/%E6%95%99%E7%A8%8B%EF%BC%9A5%E5%88%86%E9%92%9F%E4%B8%8A%E6%89%8BXCharts.md)里说的开启TMP即可，内部会自动加上宏和在asmdef加上TextmeshPro的引用。

如果还报错，需要自己手动处理以上两点。

**Comment by WKC23113** [comment]
您好，请问可以使用代码更改折线图、饼图的颜色以及样式吗？

**Reply by monitor1394** [reply]
可以，editor上看到的所有配置项都可以用代码动态更改。serie和数据也可以代码动态添加和更新。

**Reply by WKC23113** [reply]
我查了XChartsAPI，但是没有找到，您可以给我些提示吗？

**Reply by monitor1394** [reply]
建议先看五分钟教程，试用熟悉一下先。

**Reply by WKC23113** [reply]
我已经在用了 但是没有找到 所以才来问您😁

**Reply by monitor1394** [reply]
如果你了解是哪些组件控制了对应的效果，就可以获取到组件，改里面的参数。这个五分钟教程里写的很清楚了。
所以，重点在于你怎么定位你要改的效果对应什么组件。
比如，折线图颜色，可以通过改serie的lineStyle的color。
其他细节没法一一说明了，需要自己去熟悉。
有需要可以加QQ群在里面交流。

**Reply by WKC23113** [reply]
好的明白了，非常感谢
